### PR TITLE
Disallow removal of allocated nodes

### PR DESCRIPTION
**Pull request by mpolden**
First commit is the interesting one.

FYI @tokle

### CODE REVIEW

**Comment by mpolden** [comment]
@bratseth ping

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java [hunk 1]
```diff
@@ -544,13 +544,15 @@ public Node markNodeAvailableForNewAllocation(String hostname, Agent agent, Stri
...
         try (Mutex lock = lockUnallocated()) {
-            List<Node> removed = !node.type().isDockerHost() ?
-                    new ArrayList<>() :
-                    getChildNodes(node.hostname()).stream()
-                            .filter(child -> force || verifyRemovalIsAllowed(child, true))
-                            .collect(Collectors.toList());
+            List<Node> removed = new ArrayList<>();
 
-            if (force || verifyRemovalIsAllowed(node, false)) removed.add(node);
+             if (node.type().isDockerHost()) {
```
Comment (bratseth) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/NodeRepository.java]: Much better!

**Review by bratseth — APPROVED** [review]
Code is much better but I don't see that you are really changing any behavior here?

**Comment by mpolden** [comment]
We now disallow removal of failed tenant nodes that are still allocated (first line of `canRemove`). This has caused a few issues in the past weeks, where deleted nodes are still referenced elsewhere in the system.

**Comment by bratseth** [comment]
Ah, right. Ok

**Pull request merged**